rust/protocol/src/kdf.rs

```diff
@@ -16,17 +18,18 @@ pub struct HKDF {
 impl HKDF {
     const HASH_OUTPUT_SIZE: usize = 32;
 
-    pub fn new(message_version: u32) -> Result<Self> {
+    pub fn new() -> Result<Self> {
+        Self::new_for_version(MessageVersion::default())
+    }
+
+    pub fn new_for_version(message_version: MessageVersion) -> Result<Self> {
         match message_version {
-            2 => Ok(HKDF {
+            MessageVersion::Version2 => Ok(HKDF {
                 iteration_start_offset: 0,
             }),
-            3 => Ok(HKDF {
+            MessageVersion::Version3 => Ok(HKDF {
                 iteration_start_offset: 1,
             }),
-            _ => Err(SignalProtocolError::UnrecognizedMessageVersion(
-                message_version,
-            )),
         }
```

Comment:
I think there's more going on here than just the message version at this point, since we also use KDF for other functionality besides just the session state. In addition, we don't actually support any message versions but 3 in this library (and the clients don't use HKDF for "other functionality" with any version but 3), so generalizing over message versions isn't buying us anything at the moment. Finally, we wouldn't want adding a new message version to change the derived keys for that "other functionality", so I'd consider a no-args constructor an anti-pattern here.

All of this suggests a redesign, either to have a proper enum here that isn't strictly about message versions, or to only support version 3 here and drop the generality. Both of those should wait for another PR.

Reply:
Ok, I've interpreted this as:
1. remove the no-arg constructor `HKDF::new()` (done).
1. split off a separate PR to remove the `MessageVersion` enum.

Is that correct?